### DISCUSSION TITLE
Getting error 'Class Illuminate\Support\Reflector not found' while calling livewire class from route

### DISCUSSION THREAD

**Discussion by jamesbhatta** [livewire/livewire#1967] [1. Help]
I am trying to directly render livewire component from routes without controller in laravel 7.

Livewire component class:

```
class MyOrders extends Component
{
    public $orders;

    public function mount()
    {
        $this->orders = auth()->user()->orders;
    }

    public function render()
    {
        return view('livewire.my-orders')
        ->extends('layouts.account')
        ->section('main');
    }
}

```
I created a separate route file having no namespace  and define route as `Route::get('my-orders', App\Http\Livewire\MyOrders::class);`. As you can see I am not want to passing any parameters to my component and want 
to declare `$orders` as the public property so that it can be used in blade without passing explicitly in render method. 

Error upon visiting the route: `Class 'Illuminate\Support\Reflector' not found` (my laravel version is 7.0 and there is no  Reflector class in this path)

A quick fix for this issue is to declare `$orders` as protected property and pass it to view as 
```
return view('livewire.my-orders', ['orders' => $this->orders])
        ->extends('layouts.account')
        ->section('main');
```

This is something that I never love to do with blade as well as livewire components. I found it very convenient to simply declare the property as public and access it in blade.

**Comment by iksaku** [comment]
Did you follow the [update instructions regarding Laravel 7](https://laravel-livewire.com/docs/2.x/upgrading#route-livewire)?

**Comment by jamesbhatta** [comment]
I haven;t read that section earlier but I have created a registered a new route file without the namespace. And its working fine but the only problem is that it is trying to bind the public properties with the parameters even though I do not have one.

As I said changing public properties to  private / protected  works fine.
